sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1414,4 +1415,37 @@ class DataFrame private[sql](
     val jrdd = rdd.map(EvaluatePython.rowToArray(_, fieldTypes)).toJavaRDD()
     SerDeUtil.javaToPython(jrdd)
   }
+
+  /////////////////////////////////////////////////////////////////////////////
+  // Statistic functions
+  /////////////////////////////////////////////////////////////////////////////
+
+  // scalastyle:off
+  object stat {
+  // scalastyle:on
+
+    /**
+     * Finding frequent items for columns, possibly with false positives. Using the algorithm
+     * described in `http://www.cs.umd.edu/~samir/498/karp.pdf`.
+     *
+     * @param cols the names of the columns to search frequent items in
+     * @param support The minimum frequency for an item to be considered `frequent`
+     * @return A Local DataFrame with the Array of frequent items for each column.
+     */
+    def freqItems(cols: Array[String], support: Double): DataFrame = {
```

Comment:
in df we usually support List[String] and Seq[String]. This is one reason why we are using a separate name space.